ext/native/gfx_es2/draw_text_android.cpp

```diff
@@ -0,0 +1,233 @@
+#include "base/display.h"
+#include "base/logging.h"
+#include "base/stringutil.h"
+#include "thin3d/thin3d.h"
+#include "util/hash/hash.h"
+#include "util/text/wrap_text.h"
+#include "util/text/utf8.h"
+#include "gfx_es2/draw_text.h"
+#include "gfx_es2/draw_text_android.h"
+
+#include "android/jni/app-android.h"
+#include <assert.h>
+
+#if PPSSPP_PLATFORM(ANDROID)
+
+#include <jni.h>
+
+TextDrawerAndroid::TextDrawerAndroid(Draw::DrawContext *draw) : TextDrawer(draw) {
+	env_ = jniEnvGraphics;
+	const char *textRendererClassName = "org/ppsspp/ppsspp/TextRenderer";
+	jclass localClass = env_->FindClass(textRendererClassName);
+	cls_textRenderer = reinterpret_cast<jclass>(env_->NewGlobalRef(localClass));
+	ILOG("cls_textRender: %p", cls_textRenderer);
+	if (cls_textRenderer) {
+		method_measureText = env_->GetStaticMethodID(cls_textRenderer, "measureText", "(Ljava/lang/String;D)I");
+		ILOG("method_measureText: %p", method_measureText);
+		method_renderText = env_->GetStaticMethodID(cls_textRenderer, "renderText", "(Ljava/lang/String;D)[I");
+		ILOG("method_renderText: %p", method_renderText);
+	} else {
+		ELOG("Failed to find class: '%s'", textRendererClassName);
+	}
+	curSize_ = 12;
+	dpiScale_ = 1.0f;
+}
+
+TextDrawerAndroid::~TextDrawerAndroid() {
+	// Not sure why we can't do this but it crashes.
+	// At worst we leak one ref...
+	// env_->DeleteGlobalRef(cls_textRenderer);
+	ClearCache();
+}
+
+bool TextDrawerAndroid::IsReady() const {
+	return cls_textRenderer != nullptr && method_measureText != nullptr && method_renderText != nullptr;
+}
+
+uint32_t TextDrawerAndroid::SetFont(const char *fontName, int size, int flags) {
+	// We will only use the default font
+	uint32_t fontHash = 0; //hash::Fletcher((const uint8_t *)fontName, strlen(fontName));
```

Comment:
Hopefully it'll have the sense to fall back to regular Roboto for missing characters as I don't think our TTF is very complete.

I'll take a stab later, would be nice to keep the current look for most languages at least.